### PR TITLE
API support for 800/1000/1020 for deploymentplan

### PR DESCRIPTION
**Pull request by chebroluharika**
**Description**
API support for API800,1000,1200 for deploymentplan

**Issues Resolved**
NA

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
  - [ ] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [x] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: endpoints-support.md [hunk 1]
```diff
@@ -24,6 +24,12 @@
 |<sub>/rest/connection-templates/defaultConnectionTemplate</sub>                          |GET          | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
 |<sub>/rest/connection-templates/{id}</sub>                                               |GET          | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
 |<sub>/rest/connection-templates/{id}</sub>                                               |PUT          | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|     **Deployment Plans**                                                                                         |
```
Comment (AnushaY1916) [inline endpoints-support.md]: Image streamer resources should be placed separately. Please refer to Ruby SDK.

File: libraries/resource_providers/image_streamer/api800.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+module OneviewCookbook
+  module ImageStreamer
+    # Module for Image Streamer API 800 Resources
+    module API800
+      # Get resource class that matches the type given
+      # @param [String] type Name of the desired class type
+      # @param [String] variant Variant is ignored
+      # @return [Class] Resource class or nil if not found
+      def self.provider_named(type, _variant = nil)
```
Comment (AnushaY1916) [inline libraries/resource_providers/image_streamer/api800.rb:22]: There should be a class inside the module and later function(def) inside the class. Is class missing here?

**Review by AnushaY1916 — CHANGES_REQUESTED** [review]

File: spec/fixtures/cookbooks/image_streamer_test_api1200.rb/attributes/default.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#
```
Comment (soodpr) [inline spec/fixtures/cookbooks/image_streamer_test_api1200.rb/attributes/default.rb]: Name of the folder is incorrect here. remove ".rb from the folder name: image_streamer_test_api1200.rb"

File: endpoints-support.md (hunk 1 above)
Comment (chebroluharika) [reply endpoints-support.md]: done

File: spec/fixtures/cookbooks/image_streamer_test_api1200.rb/attributes/default.rb (hunk 3 above)
Comment (chebroluharika) [reply spec/fixtures/cookbooks/image_streamer_test_api1200.rb/attributes/default.rb]: done

File: libraries/resource_providers/image_streamer/api800.rb (hunk 2 above)
Comment (chebroluharika) [reply libraries/resource_providers/image_streamer/api800.rb:22]: Thats the structure is being followed in all other modules too.

**Review by AnushaY1916 — APPROVED** [review]

**Review by soodpr — APPROVED** [review]

**Pull request merged**